### PR TITLE
feat(api): added ban record

### PR DESCRIPTION
**Pull request by lucianHymer**
fixes https://github.com/passportxyz/passport/issues/3051

### CODE REVIEW

File: api/ceramic_cache/models.py [hunk 1]
```diff
@@ -131,3 +134,110 @@ class Revocation(models.Model):
...
+    provider = models.CharField(default="", max_length=256, db_index=True, blank=True)
+    hash = models.CharField(default="", max_length=100, db_index=True, blank=True)
+    address = EthAddressField(default="", db_index=True, blank=True)
+    end_time = models.DateTimeField(null=True, blank=True)
+    reason = models.TextField(null=True, blank=True)
+    created_at = models.DateTimeField(auto_now_add=True)
+    last_run_revoke_matching = models.DateTimeField(
+        null=True, blank=True, help_text="Last time revoke_matching_credentials was run"
+    )
+
```
Comment (nutrina) [inline api/ceramic_cache/models.py:152]: I would include a `type` field, of type `BanType`.
That would be much clearer for people to understand, and query for particular ban types (I'm also thinking about analytics, not just eng)

Comment (lucianHymer) [reply api/ceramic_cache/models.py:180]: I'd prefer to restrict it to specific "types" at the UI level to keep the code simple. But I'm cool with whatever!

File: api/ceramic_cache/models.py [hunk 2]
```diff
@@ -113,6 +119,19 @@ class Meta:
...
 
 
+class RevocationList(models.Model):
+    name = models.CharField(
+        default="", max_length=256, db_index=True, null=True, blank=True
+    )
+    description = models.TextField(default="", null=True, blank=True)
+    csv_file = models.FileField(
+        max_length=1024, null=False, blank=False, upload_to="revocation_list"
+    )
```
Comment (nutrina) [inline api/ceramic_cache/models.py:129]: Add help text, what columns are expected?

File: api/ceramic_cache/models.py [hunk 3]
```diff
@@ -129,5 +148,182 @@ class Revocation(models.Model):
...
+
+
+class BanList(models.Model):
+    name = models.CharField(
+        default="", max_length=256, db_index=True, null=True, blank=True
+    )
+    description = models.TextField(default="", null=True, blank=True)
+    csv_file = models.FileField(
+        max_length=1024, null=False, blank=False, upload_to="ban_list"
+    )
```
Comment (nutrina) [inline api/ceramic_cache/models.py:175]: Add help_text, what column sare expected

File: api/ceramic_cache/admin.py [hunk 4]
```diff
@@ -73,5 +82,235 @@ class AccountAPIKeyAdmin(ScorerModelAdmin):
...
+        for revocation_item in csv_reader:
+            proof_value = revocation_item["proof_value"]
+            ceramic_cache = CeramicCache.objects.get(proof_value=proof_value)
+            db_revocation_item = Revocation(
+                proof_value=proof_value,
+                ceramic_cache=ceramic_cache,
+                revocation_list=obj,
+            )
+            revocation_item_list.append(db_revocation_item)
+            ceramic_cache_list.append(ceramic_cache)
```
Comment (nutrina) [inline api/ceramic_cache/admin.py:154]: Remove ceramic_cache_list variable

File: api/internal/api.py [hunk 5]
```diff
@@ -70,3 +82,91 @@ def add_stamps(
...
+        )
+
+    address = unique_ids[0].split(":")[-1]
+
+    hashes = list(
+        set([c.credentialSubject.hash for c in payload if c.credentialSubject.hash])
+    )
+
+    try:
+        bans = Ban.get_bans(address=address, hashes=hashes)
```
Comment (nutrina) [inline api/internal/api.py:110]: This would not work if an entire account is banned and only `address` is set

**Pull request merged**